### PR TITLE
Code coverage

### LINKED ISSUE #378: Increase code coverage across different modules (easy ones)
🐞 **Problem**
The goal is 100% code coverage. It's not a very realistic goal because sometimes we lack the knowledge to do so. But I am listing some of the parts, where coverage can be increased easily. 

🎯 **Goal**

Try to target these files. The lines untested would be marked red [here](https://codecov.io/gh/einsteinpy/einsteinpy/tree/master/src/einsteinpy).
  - [x] `coordinates/core.py` (#394)
  - [x] `ijit.py` (#397)
  - [x] `symbolic/vector.py`
  - [x] `symbolic/tensor.py`
  - [x] `plotting/hypersurface/core.py`
  - [x] `plotting/rays/shadow.py`
  - [x] `rays/shadow.py`

💡 **Possible solutions**

  - [What is coverage](https://confluence.atlassian.com/clover/about-code-coverage-71599496.html)
  - tests are present in `src\einsteinpy\tests\test_<module name>\test_<file name>`

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

@shreyasbapat if possible, add some more!

### PR DESCRIPTION
**Pull request by mightomi**
Fixes #378 
symbolic/vector.py and symbolic/tensor.py

### CODE REVIEW

**Comment by ritzvik** [comment]
Thanks @mightomi for the efforts :)
Everything seems fine to me. One last thing before I merge. Please squash the commits into one.

**Comment by shreyasbapat** [comment]
Hey @mightomi , I request you to please squash the commits into 1 or 2 before I merge!

**Comment by mightomi** [comment]
@shreyasbapat I am sorry for taking this long, I have squashed the commit into one

File: src/einsteinpy/tests/test_symbolic/test_tensor.py [hunk 1]
```diff
@@ -127,6 +127,15 @@ def test_Tensor_repr():
     assert not "object at 0x" in machine_representation
 
 
+def test_TypeError1():
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_tensor.py]: Can you use `xfail` here? 
For reference, see PR #446

File: src/einsteinpy/tests/test_symbolic/test_tensor.py [hunk 2]
```diff
@@ -175,6 +184,18 @@ def test_check_properties():
 # tests for BaseRelativityTensor
 
 
+def test_BaseRelativityTensor_TypeError():
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_tensor.py]: Here also.

File: src/einsteinpy/tests/test_symbolic/test_vector.py [hunk 3]
```diff
@@ -25,6 +25,14 @@ def test_GenericVector_change_config_theoretical_test():
     covec = cnvec.change_config("l")  # get contravariant vector
     assert simplify(covec.tensor()[0] - (a + b * cos(th))) == 0
     assert simplify(covec.tensor()[1] - (b + a * cos(th))) == 0
+    # tests when metric is None
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_vector.py]: Please make a separate test for this and `xfail`

**Comment by ritzvik** [comment]
Merge conflicts. I knew this would happen :(

**Comment by mightomi** [comment]
After the rebase, even after commit this PR somehow remained close so I created another PR #448. Hope it does not cause much problems.

**Pull request closed without merging**